frontend/src/components/projects/LinkSamplesToStudy.tsx

```diff
@@ -0,0 +1,125 @@
+import { notifyError, notifySuccess } from "../../modules/notification/actions";
+
+const NOTIFICATION_ID = "samples-linked-to-study"
+
+export interface LinkSamplesToStudyProps {
+    open?: boolean
+    selectAll: boolean,
+    selectedItemIDs: number[]
+    totalCount: number
+    projectID: number
+    handleOk?: () => void
+    handleCancel?: () => void
+    handleSuccess?: () => void
+}
+
+export default function LinkSamplesToStudy({ open, selectAll, selectedItemIDs, totalCount, projectID, handleOk, handleCancel, handleSuccess }: LinkSamplesToStudyProps) {
+    const dispatch = useAppDispatch()
+    const studiesByID = useAppSelector(selectStudiesByID)
+    const studies = useMemo(() =>
+        getAllItems(studiesByID)
+            .filter(study => study.project_id === projectID)
+            .sort((a, b) => {
+                if (a.letter > b.letter) return 1
+                else if (a.letter < b.letter) return -1
+                else return 0
+            })
+        , [projectID, studiesByID])
+    const [study, setStudy] = useState<Study>()
+
+    const [stepsOrder, setStepsOrder] = useState<WorkflowStepOrder[]>([])
+    const [stepOrder, setStepOrder] = useState<WorkflowStepOrder>()
+    useEffect(() => {
+        if (study) {
+            dispatch(api.workflows.get(study.workflow_id)).then(response => {
+                setStepsOrder(response.data.steps_order.filter((stepOrder) => stepOrder.order >= study.start && stepOrder.order <= study.end))
+            }).catch(() => dispatch(notifyError({
+                id: NOTIFICATION_ID,
+                title: "Failed to get study",
+                description: `Failed to get study ${study}`
+            })))
+        }
+    }, [study, dispatch])
+
+    const sampleCount = selectAll ? totalCount - selectedItemIDs.length : selectedItemIDs.length
+
+    return (
+        <Modal
+            title={`Move ${sampleCount} Sample${sampleCount > 1 ? 's' : ''} to Study`}
+            open={open}
+            okButtonProps={{
+                disabled: !(sampleCount > 0 && study && stepOrder)
+            }}
+            onOk={() => {
+                if (selectedItemIDs.length > 0 && study && stepOrder) {
+                    dispatch(api.projects.addSamplesToStudy(selectedItemIDs, selectAll, projectID, study.letter, stepOrder.order)).then(
+                        () => {
+                            dispatch(notifySuccess({
+                                id: NOTIFICATION_ID,
+                                title: "Samples linked to study",
+                                description: `Successfully linked samples to study ${study.letter} at step "${stepOrder.step_name}"`
+                            }))
+                            if (handleSuccess) {
+                                handleSuccess()
+                            }
+                        },
+                        () => {
+                            dispatch(notifyError({
```

Comment:
How to do it is a more delicate issue... i would personally keep the simple notification for failure but add to the response a "failure log" that can be downloaded by the frontend. Not sure if this is the typical way to go but i am not sure a temporary model would help since it might close before the user have a time to check the content. This is just an idea we can go another way but we need to have in mind the user has little information on the validation done in the backend to queue up a sample.